core/src/test/java/org/apache/struts2/ognl/OgnlSetPossiblePropertyTest.java

```diff
@@ -0,0 +1,240 @@
+         *
+         *  NonExcludedHolder class
+         *   - field: private
+         *   - method: public
+         *  ExcludeField class
+         *   - field: private
+         *   - set method: public (but not matching with field name)
+         */
+        HolderWhoseFieldWithPublicSetterDifferentFieldName holder = setUpClass(HolderWhoseFieldWithPublicSetterDifferentFieldName.class);
+        vs.setValue("excludedField.excludedFieldString", "EXPLOITED");
+
+        assertNotEquals("EXPLOITED", holder.excludedField.excludedFieldStringInternal);
+    }
+
+
+    public static class HolderWithPublicField {
+        public ExcludedField excludedField = new ExcludedField();
+
+        public static class ExcludedField {
+            public String excludedFieldString = "defaultValue";
+        }
+    }
+
+    public static class HolderWithPublicMethod {
+        private ExcludedField excludedField = new ExcludedField();
+
+        public ExcludedField getExcludedField() {
+            return excludedField;
+        }
+
+        public static class ExcludedField {
+            private String excludedFieldString = "defaultValue";
+
+            public void setExcludedFieldString(String value) {
+                this.excludedFieldString = value;
+            }
+        }
+    }
+
+    public static class HolderWhoseFieldWithPrivateMethod {
+        private ExcludedField excludedField = new ExcludedField();
+
+
+        public ExcludedField getExcludedField() {
+            return excludedField;
+        }
+
+        public static class ExcludedField {
+            public String excludedFieldString = "defaultValue";
+
+            private void setExcludedFieldString(String value) {
+                this.excludedFieldString = value;
+            }
+        }
+    }
+
+    public static class HolderWhoseFieldWithPublicMethod {
+        private ExcludedField excludedField = new ExcludedField();
+
+
+        public ExcludedField getExcludedField() {
+            return excludedField;
+        }
+
+        public static class ExcludedField {
+            public String excludedFieldString = "defaultValue";
+
+            private void setExcludedFieldString(String value) {
+                this.excludedFieldString = value;
+            }
+        }
+    }
+
+    public static class HolderWhoseFieldWithPublicWriteMethod {
+        private ExcludedField excludedField = new ExcludedField();
+
+
+        public ExcludedField getExcludedField() {
+            return excludedField;
+        }
+
+        public static class ExcludedField {
+            private String excludedFieldString = "defaultValue";
+
+            private void setExcludedFieldString(String value) {
+                this.excludedFieldString = value;
+            }
+
+            public void setexcludedfieldstring(String value) {
+                this.excludedFieldString = value;
+            }
+        }
+    }
+
+    public static class HolderWhoseFieldWithPublicSetterDifferentFieldName {
```

Comment:
@lukaszlenart This is a PoC of the new public setter bug discovered by @jefferyxhy